torch/utils/data/dataloader.py

```diff
@@ -174,6 +190,17 @@ def __init__(self, loader):
     def __len__(self):
         return len(self.batch_sampler)
 
+    def _get_batch(self):
+        if self.timeout > 0:
+            try:
+                return self.data_queue.get(True, self.timeout)
+            except queue.Empty:
+                ex = RuntimeError('DataLoader timed out after {} seconds'.format(self.timeout))
+                ex.__cause__ = ex  # prevent exception chaining in python 3
```

Comment:
Is this an acceptable way to prevent showing exception chain?

Reply:
No idea, don't know how this works in Py2. Maybe let's just ignore it and show the chain?